### DISCUSSION TITLE
Problems after updating macros

### DISCUSSION THREAD

**Discussion by Serj121** [jschuh/klipper-macros#138] [Q&A]
I have a problem after updating your macros to a new version.
I change the parameter variable_probe_min_count: in the file [gcode_macro _km_options]
gcode: . But when measuring at the beginning of printing, the values are taken from the file bltouch.cfg andmeasures the entire bed, not the print area. What could be the problem?

And it is parked at the end of the print at the coordinates X Y (0,0). But I specified in the settings (0,200)
I attach old and new configuration files.
[New.zip](https://github.com/jschuh/klipper-macros/files/11794948/New.zip)
[Old.zip](https://github.com/jschuh/klipper-macros/files/11794949/Old.zip)

M190 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
M109 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
_PRINT_START_PHASE_INIT EXTRUDER={first_layer_temperature[initial_tool]} BED=[first_layer_bed_temperature] MESH_MIN={first_layer_print_min[0]},{first_layer_print_min[1]} MESH_MAX={first_layer_print_max[0]},{first_layer_print_max[1]} LAYERS={total_layer_count} NOZZLE_SIZE={nozzle_diameter[0]}
; Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
_PRINT_START_PHASE_PROBING
; Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).

PRINT_END

**Comment by kOLIANIKO** [comment]
Interesting problem, but I don't know how to help you.

**Comment by jschuh** [comment]
As I mentioned when you asked before, this config is invalid. The [Klipper setup instructions](https://github.com/jschuh/klipper-macros#installation) I pointed you to earlier explain how to create a working config, but here's two major issues I saw from skimming yours:

First, these lines in `printer.cfg` are invalid, because you're declaring an empty `[gcode_macro _km_options]` section:

```
[gcode_macro _km_options]
[include AAAglobals.cfg] # сдесь хранятся настройки макросов
```

Second, there's a duplicate declaration of `[gcode_macro _km_options]` in `AAAglobals.cfg` (which Klipper should be throwing an error on). And you've copied the entire contents of `globals.cfg` into that declaration, which is setting many of the configuration options to invalid values.

You need to have a single declaration of `[gcode_macro _km_options]`, and that definition should include `variable_` declarations only for options that you are changing—_not for every option in `globals.cfg`_. That's why the installation instructions provide a [template `[gcode_macro _km_options]` section](https://github.com/jschuh/klipper-macros#klipper-setup) to paste into your config.
